### PR TITLE
core(core): convert filters to TS and ng2

### PR DESCRIPTION
**Pull request by icfantv**
PLEASE DO NOT MERGE!

* convert core time filters to Typescript.
* add new ng2 pipes that wrap the core time filters.
* convert system timezone component to Typescript.
* minor code/build cleanup and housekeeping.
* update moment and moment TZ to latest.

### CODE REVIEW

File: tslint.json [hunk 1]
```diff
@@ -61,7 +61,7 @@
       "single",
       "avoid-escape"
     ],
-    "radix": true,
+    "radix": false,
```
Comment (icfantv) [inline tslint.json]: The default is base 10 so I'm not sure why the linter thinks it's necessary to have to put the radix value there so I just shut it off.

File: webpack.common.js [hunk 2]
```diff
@@ -147,10 +147,12 @@ function configure(IS_TEST) {
     ]);
   }
 
+  config.plugins.push(new webpack.ContextReplacementPlugin(/angular(\\|\/)core(\\|\/)@angular/,path.resolve(__dirname, '../src')));
```
Comment (icfantv) [inline webpack.common.js]: This gets rid of the "dependency is an expression" build warning message.

**Comment by icfantv** [comment]
@anotherchrisberry @jrsquared @zanthrash PTAL

File: app/scripts/modules/core/delivery/executions/executions.html [hunk 3]
```diff
@@ -48,7 +48,7 @@ <h3 class="filters-placeholder">
             <option value="timeBoundary">Time Boundary</option>
           </select>
         </div>
-        <div class="form-group">
+        <div class="form-group" ng-if="vm.filter.groupBy === 'name'">
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/delivery/executions/executions.html]: Not true. It's used to determine how many executions to fetch per pipeline, which determines how many pipelines are actually shown on the screen.

File: app/scripts/modules/core/widgets/systemTimezone.component.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  public template = `<span ng-bind="$ctrl.tz"></span>`;
+  public controller = function(settings: any) {
+    const zone: string = settings.defaultTimeZone || TimeFilterConstants.DEFAULT_TIMEZONE;
+    this.tz = moment.tz(moment(), zone).zoneAbbr();
+  };
+}
+
+export const SYSTEM_TIMEZONE_COMPONENT = 'spinnaker.core.widget.systemTimezone.component';
+module(SYSTEM_TIMEZONE_COMPONENT, [require('core/config/settings.js')])
+  .component('systemTimezone', new SystemTimezoneComponent());
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/widgets/systemTimezone.component.ts]: Any reason not to write this as an ng2 component?

**Comment by anotherchrisberry** [comment]
It would be nice to see the ng2 filters either used or tested.

File: app/scripts/modules/core/widgets/systemTimezone.component.ts (hunk 4 above)
Comment (icfantv) [inline app/scripts/modules/core/widgets/systemTimezone.component.ts]: no, good point.

**Comment by icfantv** [comment]
Yea, lemme look through the app tomorrow to see if there's a good place to use the pipes - may need to convert some more stuff to ng2 first.

**Comment by icfantv** [comment]
I had a think about this and I'm thinking I should actually flip the code and put the business logic in the pipes and then try and construct/inject the pipes into the ng1 side.  This shouldn't be too much work and we leave most of the infrastructure that's already there, in place.  The one thing that may be tricky is using settings.js, as is, in ng2 land.

**Comment by anotherchrisberry** [comment]
superseded by https://github.com/spinnaker/deck/pull/3527

**Pull request closed without merging**